crowbar_framework/app/models/service_object.rb

```diff
@@ -1713,4 +1714,34 @@ def active_update(proposal, inst, in_queue, bootstrap = false)
       [400, e2.message]
     end
   end
+
+  def skip_unready_nodes(bc, inst, elements)
+    skip_unready_nodes_config = Rails.application.config.experimental["skip_unready_nodes"]
+    # return if the feature is disabled
+    return elements unless skip_unready_nodes_config["enabled"]
+    logger.warn("Experimental skip_unready_nodes feature enabled")
+    logger.debug("skip_unready_nodes: enter for #{bc}:#{inst}")
+    cleaned_elements = elements.deep_dup
+    skip_unready_nodes_config["roles"].each do |role|
+      next unless elements.key?(role)
+      elements[role].each do |n|
+        # TODO: deal with clusters...how?????
+        node = Node.find_by_name(n)
+        next if node.nil?
+        # skip if nodes are on ready or crowbar_upgrade state, we dont need to do anything
+        next if ["ready", "crowbar_upgrade"].include?(node.crowbar["state"])
+        logger.warn(
+          "Node #{n} is being removed from the deployment for #{bc}:#{inst} with role #{role}"
```

Comment:
I guess 'node is being skipped until next chef-client run' does give more information.

Reply:
+1